### ISSUE TITLE
PendingDeprecationWarning instead of DeprecationWarning

### ISSUE THREAD

**Issue by RMeli** [MDAnalysis/mdanalysis#2491]
I was looking at #2472 and found out that there are two sub-packages for hydrogen bond analysis: `MDAnalysis/analysis/hbond` and `MDAnalysis/analysis/hydrogenbonds`. In `MDAnalysis/analysis/hbond` there is this warning:

```
warnings.warn(
            "This module will be deprecated in version 1.0."
            "Please use MDAnalysis.analysis.hydrogenbonds.hbond_analysis instead.",
            category=DeprecationWarning
        )
```

I think a `PendingDeprecationWarning` is more appropriate in this case.

**Comment by IAlibay** [comment]
This is code that is in progress of getting removed as part of #2443

**Comment by RMeli** [comment]
Should it be removed? The message says that it will be deprecated in version `1.0`, not removed...

**Comment by IAlibay** [comment]
That is a very good point, I hadn't read it properly it seems.

Maybe one of the coredevs can clarify here, I'm glad I hadn't pushed part 2 of my removals yet :sweat_smile:

**Comment by RMeli** [comment]
I think if we are going for `0.21` it should be changed to `PendingDeprecationWarning`, while if we are going for `1.0` we should simply change the error message to something like this:
```
warnings.warn(
            "This module is deprecated."
            "Please use MDAnalysis.analysis.hydrogenbonds.hbond_analysis instead.",
            category=DeprecationWarning
        )
```

**Comment by IAlibay** [comment]
Yes, I agree, I'll add the change assuming v1.0 as part of the removal changes.

**Comment by orbeckst** [comment]
In this case PendingDeprecationWarning seems more appropriate than DeprecationWarning. Is this an existing warning in 2 and 3?

1.0 will contain both hbonds modules as we want to make it easy for users to keep the last 2-capable version of MDA for their projects. 

For 2.0 we would want to remove hbonds — if we can.

**Comment by RMeli** [comment]
Yes, `PendingDeprecationWarning` is also [available in Python 2.7](https://docs.python.org/2/library/exceptions.html#exceptions.PendingDeprecationWarning). I can change it to `PendingDeprecationWarning` if there is a version 0.21; otherwise I'll live it as it is for 1.0.

I checked other deprecation warnings and this is the only one where "pending" is more appropriate.

**Comment by orbeckst** [comment]
Thanks for checking. It seems that we’re heading for 1.0 so leave it.
